azurelinuxagent/ga/exthandlers.py

```diff
@@ -319,9 +317,35 @@ def handle_ext_handlers(self, etag=None):
 
         self.ext_handlers.extHandlers.sort(key=operator.methodcaller('sort_key'))
         for ext_handler in self.ext_handlers.extHandlers:
-            # TODO: handle install in sequence, enable in parallel
+            self.wait_on_ext_handler_dependencies(ext_handler)
             self.handle_ext_handler(ext_handler, etag)
-    
+
+    def wait_on_ext_handler_dependencies(self, ext_handler):
```

Comment:
This method concerns me.  This may have an effect on TDP. The increase in time the agent will pay attention to an extension has gone up considerably.  I don't yet know what the impact of that will be because we don't have data to track that.

Reply:
Please add test cases for this.